contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/schema/HiveDatabaseSchema.java

```diff
@@ -72,4 +80,76 @@ public String getTypeName() {
     return HiveStoragePluginConfig.NAME;
   }
 
+  @Override
+  public List<Pair<String, ? extends Table>> getTablesByNames(final List<String> tableNames) {
+    final String schemaName = getName();
+    final List<Pair<String, ? extends Table>> tableNameToTable = Lists.newArrayList();
+    List<org.apache.hadoop.hive.metastore.api.Table> tables;
+    // Retries once if the first call to fetch the metadata fails
+    synchronized(mClient) {
+      final List<String> tableNamesWithAuth = Lists.newArrayList();
+      for(String tableName : tableNames) {
+        try {
+          if(mClient.tableExists(schemaName, tableName)) {
```

Comment:
There are two parts which makes the query slow. 
One follows from your point; The other is fetching partitions which turned out not used. [1]

[1] https://github.com/apache/drill/blob/245da9790813569c5da9404e0fc5e45cc88e22bb/contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/DrillHiveMetaStoreClient.java#L236